engines/router/missionctl/handlers/http_handler_test.go

```diff
@@ -224,6 +253,97 @@ func TestHTTPServiceBadRequest(t *testing.T) {
 	}
 }
 
+// TestLogRequestSummary checks that copyResponseToLogChannel is called with each piece of
+// info to be logged and finally, logTuringRouterRequestSummary is called.
+func TestLogRequestSummary(t *testing.T) {
+	logUtilsAllComponents := &testLogUtils{}
+	logUtilsBadEnricher := &testLogUtils{}
+
+	// Define check for goroutine called
+	checkLogRequestSummaryCalled := func(t *testing.T, l *testLogUtils) {
+		timeout := time.After(3 * time.Second)           // Wait for a maximum of 3 mins
```

Comment:
Ran this test case locally a 100 times to check that it is not flaky.

Reply:
You mean seconds?